cps/src/main/js/workflow-editor.js

```diff
@@ -48,7 +48,46 @@ jenkinsJSModules.import('ace-editor:ace-editor-122')
                 });
 
                 editor.on('blur', function() {
-                    textarea.trigger('change');
+                    editor.session.clearAnnotations();
+                    var url = textarea.attr("checkUrl") + 'Json';
+
+                    $.ajax({
+                        url: url,
+                        data: {
+                            value: editor.getValue(),
+                            sandbox: $("input[name='_.sandbox']").is(':checked')
+                        },
+                        method: textarea.attr("checkMethod") || "get",
+                        success: function(data) {
+                            data = $.parseJSON(data);
+                            var annotations = [];
+                            if (data.status) {
+                                // Top level statuses
+                                if (data.status === 'success') {
+                                    return;
+                                }
+                                if (data.status === 'approval') {
+                                    annotations.push({
+                                        row: 0,
+                                        column: 0,
+                                        text: data.message,
+                                        type: "warning"
```

Comment:
This would better be served with form validation I think. Simpler code too: the JSON response could concentrate on Groovy errors.